File: BE/XoSoKienThiet/Models/NghiepVu/LienHeModel.cs
using XoSoKienThiet.Models.Core;

namespace XoSoKienThiet.Models.NghiepVu
{
    public class LienHeModel:Audit, TEntity<string>
    {
        public string DiaChi { get; set; }
        public string SoDienThoai { get; set; }
        public string Fax { get; set; }
        public string Email { get; set; }
        public string VanPhongDaiDien { get; set; }
        public int MaSoThue { get; set; }
        public string NguoiBienTap { get; set; }
        
    }
}

File: XSKT.Admin/src/models/lienheModel.js
const toJson = (item) => {
    return {
        _id: item._id,
        name: item.name,
        diaChi: item.diaChi,
        soDienThoai: item.soDienThoai,
        fax: item.fax,
        email: item.email,
        vanPhongDaiDien: item.vanPhongDaiDien,
        maSoThue: item.maSoThue,
        nguoiBienTap: item.nguoiBienTap
    }
}

const fromJson = (item) => {
    return {
        _id: item._id,
        name: item.name,
        diaChi: item.diaChi,
        soDienThoai: item.soDienThoai,
        fax: item.fax,
        email: item.email,
        vanPhongDaiDien: item.vanPhongDaiDien,
        maSoThue: item.maSoThue,
        nguoiBienTap: item.nguoiBienTap
    }
}

const baseJson = () => {
    return {
        _id: null,
        name: null,
        diaChi:null,
        soDienThoai: null,
        fax: null,
        email:null,
        vanPhongDaiDien: null,
        maSoThue: null,
        nguoiBienTap: null
    }
}

const toListModel = (items) =>{
    if(items.length > 0){
        let data = [];
        items.map((value, index) =>{
            data.push(fromJson(value));
        })
        return data??[];
    }
    return [];
}
export const lienHeModel = {
    toJson, fromJson, baseJson, toListModel
}
